File: 9663/main.py
n = int(input())

possible = []



answ = 0

def findsafe(status, pos):
    y = len(status)
    for i in range(y):
        if pos == status[i]:
            return False
        elif pos - status[i] == y-i:
            return False
        elif pos - status[i] == i-y:
            return False
    return True


def placeq(status, recurse): # status: [7 1 4 0 2 ] like
    if recurse == n:
        return 1
    tmp = 0
    for i in range(n):
        if findsafe(status,i):
            tmparr = status[:]
            tmparr.append(i)
            tmp += placeq(tmparr, recurse+1)
    return tmp

print(placeq(possible, 0))

File: 9663/main.c
#include <stdio.h>

int n;

int findsafe(int *status, int pos, int y){
    for (int i=0;i<y;i++){
        if (pos == status[i]) return 0;
        else if (pos - status[i] == y-i) return 0;
        else if (pos - status[i] == i-y) return 0;
    }
    return 1;
}

int placeq(int *status, int recurse){
    if (recurse == n) return 1;
    int tmp= 0;
    for(int i=0;i<n;i++){
        if (findsafe(status, i, recurse)){
            status[recurse] = i;
            tmp += placeq(status, recurse+1);
        }
    }
    return tmp;
}

int main(){
    scanf("%d", &n);
    int possible[n];
    printf("%d\n", placeq(possible,0));
    return 0;
}